megamolbart/megamolbart/inference.py

```diff
@@ -174,33 +174,34 @@ def smiles2embedding(self, smiles, pad_length=None):
         torch.cuda.empty_cache()
         return embedding, pad_mask
 
-    def inverse_transform(self, embeddings, model, mem_pad_mask, k=1, sanitize=True):
+    def inverse_transform(self, embeddings, mem_pad_mask, k=1, sanitize=False):
         mem_pad_mask = mem_pad_mask.clone()
         smiles_interp_list = []
 
         batch_size = 1  # TODO: parallelize this loop as a batch
         with torch.no_grad():
             for memory in embeddings.permute(1, 0, 2):
 
-                decode_fn = partial(model._decode_fn,
+                decode_fn = partial(self.model._decode_fn,
                                     mem_pad_mask=mem_pad_mask.type(torch.LongTensor).cuda(),
                                     memory=memory)
 
-                mol_strs, log_lhs = model.sampler.beam_decode(decode_fn,
-                                                              batch_size=batch_size,
-                                                              device='cuda',
-                                                              k=k)
+                mol_strs, log_lhs = self.model.sampler.beam_decode(decode_fn,
+                                                                   batch_size=batch_size,
+                                                                   device='cuda',
+                                                                   k=k)
                 mol_strs = sum(mol_strs, [])  # flatten list
 
                 # TODO: add back sanitization and validity checking once model is trained
                 logger.warn('WARNING: MOLECULE VALIDATION AND SANITIZATION CURRENTLY DISABLED')
```

Comment:
@rilango Two quick requests on this if merging isn't urgent:
1. The logging function warning that validation has been disabled should be removed since we added it back (thanks for catching that)
2. We will need a way to disable the validation step when benchmarking models. A flag would probably be easiest, do you agree?

Reply:
'sanitize' is already in use for circumventing validation.